### DISCUSSION TITLE
Override analyzerMode via environment variable

### DISCUSSION THREAD

**Discussion by dleavitt** [webpack-contrib/webpack-bundle-analyzer#614] [Ideas]
#### Issue description

By default, webpack bundle analyzer starts a server, which means that the webpack process will never terminate.

This is undesirable in CI and other non-interactive environments - your jobs will hang indefinitely. Currently (as far as I can tell) there's no way to prevent this from happening without changing the webpack configuration code. This isn't always possible in CI (e.g. it's someone else's code, or it's an older version of your own code prior to the configuration change.)

It would be great if there was some way to override the `analyzerMode` setting from the command-line, e.g. `WEBPACK_BUNDLE_ANALYZER_MODE=json` which could be set in non-interactive environments to ensure that the process will never hang.

Happy to submit a PR if this sounds useful.

#### Technical info

<pre>
  System:
    OS: macOS 11.5.2
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 260.25 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.14.0 - ~/.nodenv/versions/16.14.0/bin/node
    Yarn: 1.22.17 - /usr/local/bin/yarn
    npm: 8.3.1 - ~/.nodenv/versions/16.14.0/bin/npm
  npmPackages:
    compression-webpack-plugin: ^10.0.0 => 10.0.0 
    css-minimizer-webpack-plugin: ^4.2.2 => 4.2.2 
    fork-ts-checker-webpack-plugin: ^8.0.0 => 8.0.0 
    terser-webpack-plugin: 5 => 5.3.7 
    webpack: 5 => 5.82.0 
    webpack-assets-manifest: 5 => 5.1.0 
    webpack-bundle-analyzer: ^4.5.0 => 4.8.0 
    webpack-cli: 5 => 5.0.2 
    webpack-dev-server: ^4.7.4 => 4.13.3 
    webpack-merge: 5 => 5.8.0 
</pre>

**Comment by valscion** [comment]
Hi! Thanks for the the suggestion. I think it's out of scope for `webpack-bundle-analyzer` to start using environment variables to pick option values but if there's a lot of need for this from others, then I might be persuaded otherwise.

If we'd go with adding environment variables support, I'd hope we'd do it for all possible options and not only the `analyzerMode` one.

**Reply by dleavitt** [reply]
Thanks @valscion, makes sense.

My inspiration was the documentation for this 1st-party Github action, which I believe will just hang:
https://github.com/github/webpack-bundlesize-compare-action#how-to-use-it

I left them a note about it here:
https://github.com/github/webpack-bundlesize-compare-action/issues/499
Does that note seem accurate?

**Reply by valscion** [reply]
I've never used https://github.com/github/webpack-bundlesize-compare-action myself but it does look like they could do with using the `--json` flag and not use `webpack-bundle-analyzer` at all. But I might be missing something.

**Reply by cmaitrepierre** [reply]
I think the same as Daniel Leavitt. As we want to replace Unit Test (karma depracated now) by Cypress component, is it very important to run the Component Testing (CT) in our CI (Gitlab) for the development target. This infinite loop can be break by gitlab : variables:
    RUNNER_SCRIPT_TIMEOUT: 5m but tests not performed and no report generated.
    
<img width="911" height="380" alt="image" src="https://github.com/user-attachments/assets/f972a733-ab6e-4aa8-89ac-73c9c4cd7cb1" />
<img width="1152" height="266" alt="image" src="https://github.com/user-attachments/assets/215a9fd4-f499-43c1-89a3-b7cf48aacb36" />
